### PR TITLE
Add RetinaNet improved weights

### LINKED ISSUE #5307: Improve the accuracy of Detection models by using SOTA recipes and primitives
### 🚀 The feature

Similar to #3995 but focus on Object Detection.

Here is the accuracy of `fasterrcnn_resnet50_fpn` on latest main:
```
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.369
0:  Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.585
0:  Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.396
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.212
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.403
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.482
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.307
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.485
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.509
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.317
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.544
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.649
```

Some of the recent ML techniques added for the SSD work can immediately improve the accuracy of the FasterRCNN models without any additional changes. Obviously much bigger improvements can be achieved by carefully tuning our training recipes to the needs of each model. Using the SSD data augmentation on `fasterrcnn_resnet50_fpn` we can improve the accuracy by 2.1 mAP:
```
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.390
0:  Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.601
0:  Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.422
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.238
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.424
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.500
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.323
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.520
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.547
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.374
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.587
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.681
```

### Motivation, pitch

Kick-off a Batteries Included phase 2 project that will focus on improving object detection. After adding the necessary primitives, create a new recipe that improves the accuracy of existing models and retrain them to offer better weights to the community.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by datumbox**
Fixes #5307

Adds new pre-trained weights for RetinaNet + ResNet50 + FPN for the v2 variant with post-paper optimizations (no FrozenBN + c5 instead of p5 input on extra layers + GN on Head + gIoU loss). It improves the previous baseline by +5.1 mAP.

Trained with:
```
python -u run_with_submitit.py --ngpus 8 --nodes 1 --weights-backbone ResNet50_Weights.IMAGENET1K_V2 \ 
--dataset coco --model retinanet_resnet50_fpn_v2 --opt adamw --lr 0.0001 --epochs 26 --lr-steps 16 22 \
--weight-decay 0.05 --norm-weight-decay 0.0 --data-augmentation multiscale --sync-bn
```

Verified with:
```
torchrun --nproc_per_node=1 train.py --test-only --weights RetinaNet_ResNet50_FPN_V2_Weights.COCO_V1 \
--model retinanet_resnet50_fpn_v2 -b 1

IoU metric: bbox
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.415
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.618
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.439
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.270
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.457
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.538
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.337
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.543
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.587
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.418
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.629
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.721
```

### CODE REVIEW

File: torchvision/models/detection/retinanet.py [hunk 1]
```diff
@@ -691,7 +696,17 @@ class RetinaNet_ResNet50_FPN_Weights(WeightsEnum):
 
 
 class RetinaNet_ResNet50_FPN_V2_Weights(WeightsEnum):
-    pass
+    COCO_V1 = Weights(
+        url="",
+        transforms=ObjectDetection,
+        meta={
+            **_COMMON_META,
+            "num_params": 38198935,
```
Comment (YosuaMichael) [inline torchvision/models/detection/retinanet.py:704]: Question: Since the architecture comes from paper `Bridging the Gap Between Anchor-based and Anchor-free Detection via Adaptive Training Sample Selection` that published in 2019, shouldn't we have `publication_year: 2019` here?

Comment (datumbox) [reply torchvision/models/detection/retinanet.py:704]: This is a good point. The specific paper doesn't really describe the retinanet architecture but rather proposes a few optimizations to close the gap between anchor-based and anchor-free architectures. So a very good question is, what the `publication_year` should contain? Somehow I felt that it should still refer to the original paper but no strong opinion, happy to change it.

Another possibility is to remove completely the `publication_year` from all meta-data. Some background info on why it's there: I've added it few months back when I was doing some analysis on the volumes of the model weights downloads from the S3 bucket to understand which models we should focus on improving. Back then on my stats script I was also checking how old the model is hence why this exists in here. But perhaps it creates more issues than is fixes; the paper is missing after all. Thoughts?

Comment (datumbox) [reply torchvision/models/detection/retinanet.py:705]: I'm going to update the publication_year here to unblock but let's create an issue to remove the `publication_year` field from all meta-data. I'm OK to review other ways to encode this, but this single numeric field is not the right way to do it. What happens if there are multiple papers?

**Comment by YosuaMichael** [comment]
LGTM

**Review by YosuaMichael — APPROVED** [review]
LGTM!

**Comment by jjeremy40** [comment]
Hi, 
Where can I found that run_with_submitit.py file ?
Thanks.

**Comment by NicolasHug** [comment]
We use a slightly modified version of https://github.com/facebookresearch/deit/blob/main/run_with_submitit.py
